src/FastExpressionCompiler.LightExpression/Expression.cs

```diff
@@ -949,6 +971,46 @@ internal GotoExpression(GotoExpressionKind kind, LabelTarget target, Expression
         }
     }
 
+    public class SwitchCase
+    {
+        public readonly ReadOnlyCollection<Expression> TestValues;
+        public readonly Expression Body;
+
+        public System.Linq.Expressions.SwitchCase ToSwitchCase() =>
+            SysExpr.SwitchCase(Body.ToExpression(), TestValues.Select(x => x.ToExpression()));
+
+        public SwitchCase(Expression body, IEnumerable<Expression> testValues)
+        {
+            Body = body;
+            TestValues = new ReadOnlyCollection<Expression>(testValues.ToArray());
```

Comment:
Too many allocations.
Instead of `ReadOnlyCollection` for the field, please use `IReadOnlyList`.
Instead of `ToArray` please use `Tools.AsReadOnlyList`.

Reply:
If I change ReadOnlyCollection to IReadOnlyList we are not directly Code compatible, cause in Microsoft Code it's a IReadOnlyList

Reply:
It is fine, it will be an API surface compatible.
The LE.Expression will not be the full implementation compatible with System Expression anyway. The goal is to have an optimized implementation as much as possible.